File: 0007-reverse-integer/0007-reverse-integer.py
MAX_VAL = 2 ** 31 - 1
MIN_VAL = -1 * (2 ** 31)

class Solution:
    def reverse(self, x: int) -> int:
        multiplier = 1
        if x < 0:
            x *= -1
            multiplier = -1
        
        result = 0
        while x > 0:
            result = (result * 10) + (x % 10)
            if result > MAX_VAL or result < MIN_VAL:
                return 0
            
            x -= x % 10
            x /= 10
        
        return int(result) * multiplier

    """
class Solution:
    def reverse(self, x: int) -> int:
        max_val = 2 ** 31 - 1
        min_val = -1 * (2 ** 31)
        
        if x < 0:
               """ 

File: 0007-reverse-integer/0007-reverse-integer.go
import (
    "math"
)

func reverse(x int) int {
    multiplier := 1
    if x < 0 {
        x *= -1
        multiplier = -1
    }
    
    result := 0
    for x > 0 {
        result = (result * 10) + (x % 10)
        x -= x % 10
        x /= 10
    }
    
    result *= multiplier
    if result < math.MinInt32 || math.MaxInt32 < result {
        return 0
    }
    
    return result 
}